squape/object_tree.py

```diff
@@ -269,3 +271,85 @@ def _get_object_reference(object_or_name: any) -> any:
         # Symbolic name
         return squish.waitForObjectExists(object_or_name)
     return object_or_name
+
+
+@objectNotFoundDebugging(False)
+def _wait_for_any_object(
+    lookup_function, object_names: list, timeout: int, retry_delay: float
+):
+    lookup_errors = set()
+    start_time = time.time()
+
+    while True:
+        for obj_name in object_names:
+            try:
+                return lookup_function(obj_name, 0)
+
+            except LookupError as e:
+                lookup_errors.add(str(e))
+
+        elapsed_time = time.time() - start_time
+        if elapsed_time > timeout:
+            break
```

Comment:
How about putting that into the while condition?

Reply:
Improved in [3bdb214](https://github.com/CyberAlpaca/squish-api-python-extension/pull/52/commits/3bdb21477d598d8af494821bc0a1f0f0decd2ab5)

Reply:
Improved even more with [633ac11](https://github.com/CyberAlpaca/squish-api-python-extension/pull/52/commits/633ac11ca5395e30b8aa9686f0245825807255c1)